R/apa_print_manova.R

```diff
@@ -0,0 +1,115 @@
+
+#' @param test For MANOVA, the multivariate test statistic to be reported, see \code{\link{summary.manova}}.
+#' @rdname apa_print.aov
+#' @export
+
+apa_print.manova <- function(x, test = "Pillai", ...) {
+
+  tidied_x <- as.data.frame(
+    x = broom::tidy(x, test = test, ...)
+    , stringsAsFactors = FALSE
+  )
+  print_manova(x = tidied_x, ...)
+}
+
+
+#' @rdname apa_print.aov
+#' @export
+
+apa_print.summary.manova <- function(x, ...) {
+
+  tidied_x <- as.data.frame(
+    x = broom::tidy(x, ...)
+    , stringsAsFactors = FALSE
+  )
+  print_manova(x = tidied_x, ...)
+}
+
+
+
+#' @keywords internal
+
+print_manova <- function(
+  x
+  , in_paren = FALSE
+) {
+
+
+
+  estimate_names <- c(
+    pillai = "V"
+    , wilks = "\\Lambda"
+    , hl = "T"
+    , roy = "\\theta"
+  )
+
+
+  estimate_names_with_dollars <- paste0("$", estimate_names, "$")
+  names(estimate_names_with_dollars) <- names(estimate_names)
+
+  multivariate_stat <- intersect(names(estimate_names), colnames(x))
+
+  if(length(multivariate_stat) == 0L) {
+    stop("No multivariate test statistic found in results object.")
+  } # move this to input validation
+
+  col_names <- c(
+    term = "Effect"
+    , df = "df"
+    , pillai = "pillai"
+    , wilks = "wilks"
+    , hl = "hl"
+    , roy = "roy"
+    , statistic = "F"
+    , num.df ="df1"
+    , den.df = "df2"
+    , p.value ="p"
+  )
+  colnames(x) <- col_names[colnames(x)]
+
+  var_labels <- c(
+    estimate_names_with_dollars
+    , "Effect" = "Effect"
+    , "F" = "$F$"
+    , "df1" = "$\\mathit{df}_1$"
+    , "df2" = "$\\mathit{df}_2$"
+    , "p" = "$p$"
+  )
+
+  x <- x[x$Effect != "Residuals", colnames(x) != "df", drop = FALSE]
+
+  x[["Effect"]] <- prettify_terms(x[["Effect"]])
+  x[[multivariate_stat]] <- printnum(x[[multivariate_stat]])
+  x[["F"]] <- printnum(x[["F"]])
+  x[["df1"]] <-as.character(round(x[["df1"]], digits = 2))
+  x[["df2"]] <-as.character(round(x[["df2"]], digits = 2))
+  x[["p"]] <- printp(x[["p"]])
+
+  x <- sort_terms(x, "Effect")
+
+  variable_labels(x) <- var_labels[colnames(x)]
+
+  # assemble final output object
+  res <- apa_print_container()
+
+  res$statistic <- as.list(apply(X = x, MARGIN = 1, function(y) {
+    stat <- paste0(
+      "$", estimate_names[[multivariate_stat]], " = ", y[[multivariate_stat]], "$, "
+      , "$F(", y[["df1"]], ", ", y[["df2"]], ") = ", y[["F"]], "$, "
+      , "$p ", add_equals(y[["p"]]), "$"
+    )
+    stat
+  }))
+  if(in_paren) res$statistic <- in_paren(res$statistic)
+  names(res$statistic) <- gsub(x$Effect, pattern = " $\\times$ ", replacement = "_", fixed = TRUE)
```

Comment:
Any particular reason you are not using `sanitize_terms()` on the original effect names here?

Reply:
not aware of sanitize_terms... :laughing: